src/Components/Server/src/Circuits/RemoteUriHelper.cs

```diff
@@ -14,34 +14,49 @@ namespace Microsoft.AspNetCore.Components.Server.Circuits
     /// </summary>
     public class RemoteUriHelper : UriHelperBase
     {
-        private readonly IJSRuntime _jsRuntime;
+        private IJSRuntime _jsRuntime;
 
-        /// <summary>
-        /// Creates a new <see cref="RemoteUriHelper"/>.
-        /// </summary>
-        /// <param name="jsRuntime"></param>
-        public RemoteUriHelper(IJSRuntime jsRuntime)
+        public RemoteUriHelper()
         {
-            _jsRuntime = jsRuntime;
         }
 
         /// <summary>
         /// Initializes the <see cref="RemoteUriHelper"/>.
         /// </summary>
         /// <param name="uriAbsolute">The absolute URI of the current page.</param>
         /// <param name="baseUriAbsolute">The absolute base URI of the current page.</param>
+        /// <param name="jsRuntime">The <see cref="IJSRuntime"/> to use for interoperability.</param>
         public void Initialize(string uriAbsolute, string baseUriAbsolute)
         {
             SetAbsoluteBaseUri(baseUriAbsolute);
             SetAbsoluteUri(uriAbsolute);
             TriggerOnLocationChanged();
+        }
+
+        /// <summary>
+        /// Initializes the <see cref="RemoteUriHelper"/>.
+        /// </summary>
+        /// <param name="uriAbsolute">The absolute URI of the current page.</param>
+        /// <param name="baseUriAbsolute">The absolute base URI of the current page.</param>
+        /// <param name="jsRuntime">The <see cref="IJSRuntime"/> to use for interoperability.</param>
+        public void Initialize(string uriAbsolute, string baseUriAbsolute, IJSRuntime jsRuntime)
```

Comment:
When we later implement support for reconnecting to an existing circuit after prerendering, will the logic here change to account for the fact that we start without a JS runtime, but then obtain one later? Will we call both `Initialize` methods on the same instance? If so, might be worth considering switching it about a bit so there's just one `Initialize`, and then separately an `AttachToJsRuntime`.

Not something to worry about right now though, since it's not yet a supported scenario. Have I understood this right?

Reply:
Yeah, that's how I see it working.

Reply:
@pranavkm has a different solution to this in his PR, so we're going to have to rationalize this at some point.

Reply:
I'll take a look at his PR, but that's fine.